includes/helper/class-wp-job-manager-helper.php

```diff
@@ -194,65 +194,148 @@ public function addon_core_version_check( $do_check, $minimum_required_core_vers
+			// Set any errors that might have occurred.
+			$replacements = $meta + [
+				'plugin_name' => $plugin_data[ $plugin_slug ]['plugin_name'] ?? __( 'your WP Job Manager plugin', 'wp-job-manager' ),
+				'plugin_url'  => $plugin['url'] ?? null,
+			];
+
+			$this->set_licensed_plugin_errors( $plugin_slug, $plugin['errors'], $replacements );
 		}
+	}
 
-		return false;
+	/**
+	 * Set the plugin license errors.
+	 *
+	 * @param string $plugin_slug The plugin slug.
+	 * @param array  $error_keys  The error keys.
+	 * @param array  $replacements The replacements.
+	 */
+	private function set_licensed_plugin_errors( $plugin_slug, $error_keys, $replacements ) {
+		$plugin_name = null;
+
+		$fallback_url   = 'https://wpjobmanager.com/';
+		$plugin_name    = $replacements['plugin_name'];
+		$purchase_url   = $replacements['plugin_url'] ?? $fallback_url;
+		$my_account_url = $replacements['my_account_url'] ?? $fallback_url;
+
+		$errors = [];
+		if ( in_array( 'no_activation', $error_keys, true ) ) {
+			// translators: First placeholder is the plugin name, second placeholder is the My Account URL.
+			$errors['no_activation'] = sprintf( __( '<strong>Error:</strong> The license for <strong>%1$s</strong> is not activated on this website and has been removed. Manage your activations on your <a href="%2$s" rel="noopener noreferrer" target="_blank">My Account page</a>.', 'wp-job-manager' ), $plugin_name, $my_account_url );
+
+			$this->deactivate_licence( $plugin_slug, true );
+		} elseif ( in_array( 'invalid_license', $error_keys, true ) ) {
+			// translators: First placeholder is the plugin name; second placeholder is the URL to purchase the plugin.
+			$errors['invalid_license'] = sprintf( __( '<strong>Error:</strong> The license for <strong>%1$s</strong> is not valid and has been removed. <a href="%2$s" rel="noopener noreferrer" target="_blank">Purchase a new license</a> to receive updates and support.', 'wp-job-manager' ), $plugin_name, $purchase_url );
+
+			$this->deactivate_licence( $plugin_slug, true );
+		} elseif ( in_array( 'expired_key', $error_keys, true ) ) {
+			// translators: First placeholder is the plugin name, second placeholder is the My Account URL.
+			$errors['expired_key'] = sprintf( __( '<strong>Error:</strong> The license for <strong>%1$s</strong> has expired. You must <a href="%2$s" rel="noopener noreferrer" target="_blank">renew your license</a> to receive updates and support.', 'wp-job-manager' ), $plugin_name, $my_account_url );
+		} elseif ( in_array( 'expiring_soon', $error_keys, true ) ) {
+			// translators: First placeholder is the plugin name, second placeholder is the My Account URL.
+			$errors['expiring_soon'] = sprintf( __( '<strong>Error:</strong> The license for <strong>%1$s</strong> is expiring soon. Please <a href="%2$s" rel="noopener noreferrer" target="_blank">renew your license</a> to continue receiving updates and support.', 'wp-job-manager' ), $plugin_name, $my_account_url );
```

Comment:
This isn't implemented yet on WPJobManager.com but thought I'd throw it in as a nice-to-have eventually.

Reply:
Maybe we should create a ticket referencing this part then?

Reply:
Good idea: https://github.com/Automattic/wpjobmanager.com/issues/570